protocol/protocol.proto

```diff
@@ -758,6 +758,26 @@ message AddStreamToSyncRequest {
 
 message AddStreamToSyncResponse {}
 
+message AddStreamsToSyncRequest {
+  // sync_id is the id of the sync session.
+  string sync_id = 1;
+  // sync_pos is the list of streams and positions in those streams to receive updates from.
+  repeated AddStreamToSyncRequest streams = 2;
```

Comment:
It seems a bit redundant to reuse entire proto here (i.e. sync_id is not used in this case). Maybe something like this:
```suggestion
  repeated SyncCookie streams_to_add = 2;
  repeated bytes streams_to_remove = 3;
```